reportportal_client/core/test_manager.py

```diff
@@ -0,0 +1,221 @@
+            attributes = dict_to_payload(attributes)
+        if parameters:
+            parameters = dict_to_payload(parameters)
+
+        item_data = {
+            "description": description,
+            "attributes": attributes,
+            "parameters": parameters,
+            "hasStats": has_stats
+        }
+        kwargs and item_data.update(kwargs)
+        uuid = generate_uuid()
+        if not parent_item_id:
+            test_item = RPRootTestItem(self.rp_url,
+                                       self.session,
+                                       self.api_version,
+                                       self.project_name,
+                                       name,
+                                       item_type,
+                                       self.launch_id,
+                                       uuid,
+                                       **item_data)
+            self.__storage.append(test_item)
+        else:
+            parent_item = self.get_test_item(parent_item_id)
+            test_item = RPChildTestItem(self.rp_url,
+                                        self.session,
+                                        self.api_version,
+                                        self.project_name,
+                                        parent_item,
+                                        name,
+                                        item_type,
+                                        self.launch_id,
+                                        uuid,
+                                        **item_data)
+        test_item.start(start_time)
+        return uuid
+
+    def update_test_item(self, item_uuid, attributes=None, description=None,
+                         **kwargs):
+        """Update existing test item at the Report Portal.
+
+        :param str item_uuid:   test item UUID returned on the item start
+        :param str description: test item description
+        :param dict attributes: test item attributes(tags)
+                                Pairs of key and value (see class doc string)
+        """
+        self.get_test_item(item_uuid)
+        raise NotImplementedError()
+
+    def finish_test_item(self,
+                         item_uuid,
+                         end_time,
+                         status,
+                         issue=None,
+                         attributes=None,
+                         **kwargs):
+        """Finish test item.
+
+        :param item_uuid:  id of the test item
+        :param end_time:   time in UTC format
+        :param status:     status of the test
+        :param issue:      description of an issue
+        :param attributes: dict with attributes
+        :param kwargs:     other parameters
+        """
+        # check if the test is skipped, if not - do not mark as TO INVESTIGATE
+        if issue is None and status == "SKIPPED":
+            issue = {"issue_type": "NOT_ISSUE"}
+        if attributes and isinstance(attributes, dict):
```

Comment:
May it cause problem if "attributes" will be an empty dict?

Reply:
At least additional unneeded operations will be performed for an empty dict in this situation. Also, dict_to_payload() work with an empty dict was not tested.